### DISCUSSION TITLE
How to use rust from rustup?

### DISCUSSION THREAD

**Discussion by chrisands** [Homebrew/discussions#969] [Everyday usage]
Hi, I tried to install `librsvg` to install `emacs-plus@28`, but it has rust dependency. On high sierra, there is no bottle, so it builds rust from sources (and I get build error idk why). But I had installed rust using rustup, but homebrew doesn't seem to recognize it. I hard-linked rustc and cargo into `/usr/local/bin` and `/usr/local/sbin`, but it still trying to build it from sources. 
What should I do to make homebrew recognize rust existence on my mac?

**Comment by carlocab** [comment]
There's no straightforward way to tell `brew` that you already have a dependency satisfied with something that isn't installed by Homebrew. This is by design because switching out dependencies isn't tested in Homebrew CI.

Your alternatives are:

 1. `brew extract` `librsvg` and `emacs-plus@28` into your own tap and modify your own version of `emacs-plus@28` so it depends on your own `librsvg` and `librsvg` so that it doesn't depend on `rust`. You'll need to make sure `rust` is in your `PATH` when you install your own version of `librsvg`.
 2. install `librsvg`'s dependencies, then do `brew install --ignore-dependencies librsvg`. Note: doing this is fragile, and not supported.

Here are some docs on taps to help you get started:

https://docs.brew.sh/Taps
https://docs.brew.sh/Interesting-Taps-and-Forks
https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap
https://brew.sh/2020/11/18/homebrew-tap-with-bottles-uploaded-to-github-releases/
